File: pset2/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <ctype.h>
//Caesar cypher, not a safe choice but it will do.
//The program will always need an argument and only one, otherwise it will Yell at you.
//Thes argument is going to be save as an integer and used as a Key for the Cypheredtext

int main(int argc, string argv[])
{
    if (argc != 2)
    {
        printf("YEll");
        return 1;
    }
    else
    {
        int k = atoi(argv[1]);
        string s = GetString();
        for (int i = 0, n = strlen(s); i < n; i++)
            {
                if (isalpha(s[i]) || isblank(s[i]) || isgraph(s[i]))
                {
                    if (isupper(s[i]))
                    {
                        int p = abs('A' - s[i]);
                        int c = ((p + k) % 26);
                        printf("%c", c + 'A');
                    }
                    else if (islower(s[i]))
                    {
                        int p = abs('a' - s[i]);
                        int c = ((p + k) % 26);
                        printf("%c", c + 'a');
                    }
                    else
                        printf("%c", s[i]);
                }
                else
                {
                    printf("\n");
                    return 1;
                }
            }
        printf("\n");
        return 0;
    }
}

File: pset6/caesar.py
import cs50
import sys


if len(sys.argv) != 2:
    print("missing command-line argument")
    exit(1)

else:
    print("s: ", end="")
    s = cs50.get_string()
    if s == None:
        exit(1)
        
    for i in range(len(s)):
        if(s[i].isalpha()):
            if(s[i].isupper()):
                c = chr(((abs(ord('A') - ord(s[i])) + int(sys.argv[1])) % 26) + ord('A'))
                print(c, end="")
            elif(s[i].islower()):
                c = chr(((abs(ord('a') - ord(s[i])) + int(sys.argv[1])) % 26) + ord('a'))
                print(c, end="")
            else:
                print("{}".format(s[i]))
    print()
    exit(0)